### DISCUSSION TITLE
Copying a few terabytes of files

### DISCUSSION THREAD

**Discussion by cigolpl** [seaweedfs/seaweedfs#4741] [General]
Hello,

What is the most efficient way to copy several hundred million files or a few terabytes of data from SeaWeedFS to a standard filesystem? 

Currently, I am using the script below on linux, but it's quite slow. Given my use case, it would take an incredibly long time:

```bash
#!/bin/bash
copy_file() {
  filename=$1
  rclone copy sea:my_bucket/"$filename" ./my_directory/
}

# Export the function so it can be used by xargs
export -f copy_file

# Run the copy_file function on each line of the file in parallel
cat my_generated_filenames | xargs -n 1 -P 10 -I {} bash -c 'copy_file "{}"'`
```

**Comment by chrislusf** [comment]
Use "weed filer.copy"
